test/ui/test/html-report.spec.ts

```diff
@@ -1,54 +1,68 @@
-import { resolve } from 'node:path'
-import { beforeAll, describe, expect, it } from 'vitest'
-import { browserErrors, isWindows, page, ports, startServerCommand, untilUpdated } from '../setup'
+import { expect, test } from '@playwright/test'
+import type { ExecaChildProcess } from 'execa'
+import { execa } from 'execa'
 
-import { runVitest } from '../../test-utils'
+const port = 9001
+const pageUrl = `http://localhost:${port}/`
 
-const root = resolve(__dirname, '../fixtures')
-const port = ports.report
+test.describe('html report', () => {
+  let subProcess: ExecaChildProcess
+  let subProcessExit: Promise<void>
 
-// TODO: fix flakyness on windows
-describe.skipIf(isWindows)('html report', () => {
-  beforeAll(async () => {
-    await runVitest({ root, reporters: 'html', outputFile: 'html/index.html' })
+  test.beforeAll(async () => {
+    // generate vitest html report
+    await execa('vitest', [
+      'run',
+      '--reporter=html',
+    ], {
+      // stdio: "inherit",
+    })
+
+    // vite preview
+    subProcess = execa('vite', [
+      'preview',
+      '--outDir=html',
+      `--port=${port}`,
+      '--strict-port',
+    ], {
+      // stdio: "inherit",
+    })
```

Comment:
Would it be simpler to use `vite`'s JS API here?

```ts
import {preview} from 'vite';

let server;

beforeAll(async () => {
  ...
  server = await preview({ preview: { port, strictPort: true, outDir: 'html' } })
})

afterAll(async () => {
  await new Promise<void>((resolve, reject) => {
    server.httpServer.close(error => error ? reject(error) : resolve())
  })
})
```

Reply:
Vitest also has JS API - maybe use that one too? There are some utils in `test/test-utils/index.ts` for that.

Reply:
Thanks for the suggestion! I updated to use vite's preview api https://github.com/vitest-dev/vitest/pull/4721/commits/ac8439cddeb2d1d265bbc75c373c64b074c1efd8

Regarding the usage `test/test-utils/index.ts`, I was reluctant of using it since, now that playwright is responsible of transpiling typescript files, it might cause certain inconsistency since the utilities were expected to transpiled through Vite/Vitest so far.

For example, I just tested importing it, but playwright doesn't seem to support the same usage:

```ts
import { runVitest } from '../../test-utils'
```

```
> playwright test "/home/hiroshi/code/others/vitest/test/ui/test/html-report.spec.ts" "--headed"

Error: Directory import '/home/hiroshi/code/others/vitest/test/test-utils' is not supported resolving ES modules imported from /home/hiroshi/code/others/vitest/test/ui/test/html-report.spec.ts
Did you mean to import /home/hiroshi/code/others/vitest/test/test-utils/index.ts?
```